lib/markdownlint.js

```diff
@@ -895,6 +895,39 @@ function parseConfiguration(name, content, parsers) {
   };
 }
 
+/**
+ * Resolve referenced "extends" path in a configuration file
+ * using path.resolve() and require.resolve() as a fallback.
+ *
+ * @param {string} configFile Configuration file name.
+ * @param {string} referenceId Referenced path to resolve.
+ * @returns {string} resolved path to file
+ */
+function resolveConfigExtends(configFile, referenceId) {
+  const configFileDirname = path.dirname(configFile);
+  const resolvedExtendsFile = path.resolve(configFileDirname, referenceId);
+
+  try {
+    if (fs.statSync(resolvedExtendsFile).isFile()) {
+      return resolvedExtendsFile;
+    }
+  // eslint-disable-next-line unicorn/prefer-optional-catch-binding
+  } catch (error) {
+    // If fs.statSync has thrown, trying require.resolve
+  }
+
+  try {
+    return require.resolve(referenceId);
```

Comment:
```suggestion
    return require.resolve(referenceId, { paths: [configFile] });
```

This must be done on behalf of the configFile, otherwise it will be required from `markdownlint` which doesn't declare it as a dependency. Under PnP this will throw an error and under the node_modules layout makes it rely on hoisting to bring `referenceId` and `markdownlint` to the same level.

Reply:
Hi @merceyz! 👋

I tried adding that parameter in my earlier experiments, but this broke the currently tested behaviour. I need to double-check.

Reply:
It worked! 👍